langchain4j-couchbase/src/main/java/dev/langchain4j/store/embedding/couchbase/CouchbaseSearchMetadataFilterMapper.java

```diff
@@ -0,0 +1,134 @@
+package dev.langchain4j.store.embedding.couchbase;
+
+import com.couchbase.client.core.deps.com.fasterxml.jackson.databind.ObjectMapper;
+import com.couchbase.client.java.search.SearchQuery;
+import com.couchbase.client.java.search.queries.BooleanQuery;
+import com.couchbase.client.java.search.queries.MatchQuery;
+import dev.langchain4j.store.embedding.filter.Filter;
+import dev.langchain4j.store.embedding.filter.comparison.*;
+import dev.langchain4j.store.embedding.filter.logical.And;
+import dev.langchain4j.store.embedding.filter.logical.Not;
+import dev.langchain4j.store.embedding.filter.logical.Or;
+import reactor.util.annotation.NonNull;
+
+import java.util.Collection;
+
+class CouchbaseSearchMetadataFilterMapper {
```

Comment:
why are there two mappers?

Reply:
Because it would use sql++ filtering for removing embeddings and FTS query filters for finding relevant vectors.